src/persist-client/src/operators/shard_source.rs

```diff
@@ -134,8 +130,17 @@ where
     (parts, token)
 }
 
-/// Informs a `persist_source` to skip flow control on its output
-pub const NO_FLOW_CONTROL: usize = usize::MAX;
+/// Flow control configuration.
+#[derive(Debug)]
+pub struct FlowControl<'a, G: Scope> {
+    /// Stream providing in-flight frontier updates.
+    ///
+    /// As implied by its type, this stream never emits data, only progress updates.
+    // TODO: Replace `Infallible` with `!` once the latter is stabilized.
```

Comment:
super nit: empty comment line before this TODO one, please

Reply:
I wasn't sure whether the empty line should be a doc comment or a regular one. Then I noticed that in persist-client there is precedence for adding `TODO`s to doc comments, so I've made everything a doc comment now.